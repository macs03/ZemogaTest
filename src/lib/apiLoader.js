export default class ApiLoader {
  static boot(App) {
    const baseUrl = 'https://jsonplaceholder.typicode.com/';

    App.api = {
      get: (uri) => {
        return fetch(`${baseUrl}${uri}`, {
          method: 'GET',
        });
      },
      post: (uri) => {
        return fetch(`${baseUrl}${uri}`, {
          method: 'POST',
        });
      },
      delete: (uri) => {
        return fetch(`${baseUrl}${uri}`, {
          method: 'DELETE',
        });
      },
    };
  }
}
