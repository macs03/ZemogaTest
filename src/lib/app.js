import AssetsLoader from './assetsLoader';
import ApiLoader from './apiLoader';

const globalLoaders = [AssetsLoader, ApiLoader];

export default class App {
  static api = undefined;

  static boot() {
    return Promise.all(globalLoaders.map((s) => s.boot(App)))
      .then(() => {
        console.info('App is ready!!!');
      })
      .catch((err) => console.error('Error while booting the app', err));
  }
}
