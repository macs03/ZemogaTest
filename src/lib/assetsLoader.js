import {Assets} from 'react-native-ui-lib';

export default class AssetsLoader {
  static boot() {
    Assets.loadAssetsGroup('icons', {
      refresh: require('../assets/icons/refresh.png'),
      favorite: require('../assets/icons/star.png'),
      trash: require('../assets/icons/delete.png'),
      chevron: require('../assets/icons/chevronRight.png'),
    });
  }
}
