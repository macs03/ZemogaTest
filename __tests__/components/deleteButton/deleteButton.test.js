/**
 * @format
 */

import 'react-native';
import React from 'react';
import DeleteButton from '../../../src/components/deleteButton/deleteButton';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {on, Interactions, findByTestId} from '../../../src/lib/user-simulator';

it('renders correctly', () => {
  const tree = renderer.create(<DeleteButton onDelete={() => {}} />);
  expect(tree).toMatchSnapshot();
});

it('calls onDelete after tap button', () => {
  const spy = jest.fn();
  const tree = renderer.create(<DeleteButton onDelete={spy} />);

  const button = findByTestId(tree, 'deleteButton');

  on(button).simulate(Interactions.PRESS);

  expect(spy).toHaveBeenCalled();
});
