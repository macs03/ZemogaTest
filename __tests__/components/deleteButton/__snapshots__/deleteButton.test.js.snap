// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D6000C",
        "height": 80,
        "justifyContent": "center",
        "opacity": 1,
        "width": "100%",
      }
    }
    testID="deleteButton"
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
        }
      }
    >
      Delete All
    </Text>
  </View>
</View>
`;
